Sources/Afluent/SequenceOperators/OutputSequence.swift

```diff
@@ -36,6 +36,54 @@ extension AsyncSequences {
         }
     }
     
+    public struct OutputIn<Upstream: AsyncSequence & Sendable>: AsyncSequence, Sendable {
+        public typealias Element = Upstream.Element
+        
+        let upstream: Upstream
+        let range: OutputAnyIntRange
+
+        public struct AsyncIterator: AsyncIteratorProtocol {
+            var upstreamIterator: Upstream.AsyncIterator
+            let range: OutputAnyIntRange
+            var nextIndex = 0
+
+            public mutating func next() async throws -> Element? {
+                guard range.exceeds(nextIndex) == false else { return nil }
+                while let next = try await upstreamIterator.next() {
+                    if range.contains(nextIndex) {
+                        nextIndex &+= 1
+                        return next
+                    }
+                    nextIndex &+= 1
+                }
+                return nil
+            }
+        }
+
+        public func makeAsyncIterator() -> AsyncIterator {
+            AsyncIterator(upstreamIterator: upstream.makeAsyncIterator(), range: range)
+        }
+    }
+    
+    internal struct OutputAnyIntRange: Sendable {
+        /// Checks whether range conteins specifies index.
+        let contains: @Sendable (Int) -> Bool
+        /// Checks whether specified index exceeds range bounds.
+        let exceeds: @Sendable (Int) -> Bool
+        
+        static func make<R>(_ range: R) -> Self where R : RangeExpression, R.Bound == Int, R: Sendable {
+            let exceeds: @Sendable (Int) -> Bool
+            switch range {
+            case let range as Range<Int>: exceeds = { $0 >= range.upperBound }
+            case let range as PartialRangeUpTo<Int>: exceeds = { $0 >= range.upperBound }
+            case let range as PartialRangeThrough<Int>: exceeds = { $0 > range.upperBound }
+            case let range as ClosedRange<Int>: exceeds = { $0 > range.upperBound }
```

Comment:
So a potential here is that this just operates over known standard library range types. If I write my own custom RangeExpression this will fail.

I think we might just need to use `contains` and have the range be a generic on the operator. 

I'll leave a more detailed review later, but this part is my biggest feedback

Reply:
It won't fail it will just not end before the upstream.
The range is already generic.
There is kinda very little reason to use custom ranges. Built-in ranges cover all possibilities for continuous ranges. And if range is not continuous it is likely a very niche use case. What we could do is introduce a more generic `filter` api that allows ending sequence once all filter requirements are met. Then it would be used as a backbone for these `output` implementations.

Reply:
Right, but the API allows it (in Combine, too). You can see I could make a custom RangeExpression: https://developer.apple.com/documentation/combine/currentvaluesubject/output(in:)

It's even possible for me to make a completely custom but contiguous range, so your case will default to doing the wrong thing. I think the switch is problematic for that reason.

Reply:
Yes. The public API allows that too. My point is that there is no reason to make a custom implementation of a continuous range. 
I updated the PR description to include alternatives. I can make a custom logic here to end immediately if range goes from containing to not containing elements. And I can add a bool with default to indicate whether range is continuous or not. 
But I think it is a very unlikely use case. And Apple APIs don't have nay such workarounds. I could test them to see whether non-continuous range even works with Combine.

Reply:
https://forums.swift.org/t/critic-notes-about-swift-range-types-jungle/66317
some discussion about range types. It is not perfect so some ugliness is expected.